raftify/src/cli/mod.rs

```diff
@@ -15,116 +15,103 @@ use crate::{
     AbstractLogEntry, AbstractStateMachine, CustomFormatter, Result,
 };
 
+/// It used nested subcommands struct, using clap 
+/// [`Commands`] -> represent functions
+/// [`DebugSubcommands`], [`Dump`] -> these parse additional factors for commands to use.
+/// if you enter the command like `raftify-cli debug persisted-all ./logs`
+/// it's expected [`App::command`] is debug, [`DebugSubcommands::PersistedAll::path`] is./logs
+#[derive(Parser)]
+#[command(name = "raftify")]
+#[command(version = PKG_VERSION)]
+#[command(author = PKG_AUTHORS)]
+#[command(about = PKG_DESCRIPTION)]
+struct App {
+    /// required main commands
+    #[command(subcommand)]
+    command: Commands,
+}
+
+// Subcommand Select
+#[derive(Subcommand)]
+enum Commands {
+    /// Debug tools
+    #[command(subcommand)]
+    Debug(DebugSubcommands),
+    /// Dump tools
+    Dump(Dump),
+    /// plz, list the following functions
+    Nan,
+}
+
+#[derive(Subcommand)]
+enum DebugSubcommands {
+    /// List persisted log entries and metadata
+    Persisted {
+        /// The log directory path
+        path: String,
+    },
+    /// List persisted log entries and metadata for all local nodes
+    PersistedAll {
+        /// The log directory path
+        path: String,
+    },
+    ///List all log entries
+    Entries {
+        /// The address of the RaftNode
+        address: String,
+    },
+    /// Inspect RaftNode
+    Node {
+        /// The address of the RaftNode
+        address: String,
+    },
+}
+
+#[derive(Args)]
+struct Dump {
+    /// The log directory path
+    path: String,
+    /// The peers to dump
+    peers: String,
+}
+
 pub async fn cli_handler<
     LogEntry: AbstractLogEntry + Debug + Send + 'static,
     FSM: AbstractStateMachine + Debug + Clone + Send + Sync + 'static,
 >(
-    custom_args: Option<Vec<String>>,
+    _custom_args: Option<Vec<String>>,
```

Comment:
The function is also used when exposing the CLI module to the Python side. When calling `cli_handler` from the Rust side, `custom_args` is None, but when calling from the Python side, you need to pass a value to `custom_args` and parse it accordingly.

So, I suggest making the following changes

```rs
pub async fn cli_handler<
    LogEntry: AbstractLogEntry + Debug + Send + 'static,
    FSM: AbstractStateMachine + Debug + Clone + Send + Sync + 'static,
>(
    args: Option<Vec<String>>,
) -> Result<()> {
    let app = match args {
        Some(args) => App::parse_from(args),
        None => App::parse(),
    };
...
```